bin/runtime/src/lib.rs

```diff
@@ -709,14 +709,27 @@ parameter_types! {
     pub CodeHashLockupDepositPercent: Perbill = Perbill::from_percent(30);
 }
 
+// The filter for the runtime calls that are allowed to be executed by contracts.
+// Currently we allow only staking and nomination pools calls.
+pub struct ContractsCallRuntimeFilter;
```

Comment:
insignificant comment: when declaring a type that is not intended for instantiating, you can use empty enum - this way you can create a new, uninhabited type

Reply:
Is there any performance or memory difference between your suggestion and the struct?

Reply:
nope

Reply:
Right, I removed the pesky habitants :)